File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/model/Rect.java
package com.sergey.pisarev.model;

public class Rect extends Point {
    double height;
    double width;

    public Rect(){}

    public Rect(Point point, double height, double width) {
        x = point.getX();
        z = point.getZ();
        this.height = height;
        this.width = width;
    }

    public void setRect(Point point, double height, double width){
        x = point.getX();
        z = point.getZ();
        this.height = height;
        this.width = width;
    }

    public void setRect(double x,double z, double height, double width){
        this.x = x;
        this.z = z;
        this.height = height;
        this.width = width;
    }

    public boolean isInsideRect(double x, double z) {
        return (x >= this.x && x <= this.x + width) && (z >= this.z && z <= this.z + height);
    }
}

File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/model/Rect.kt
package com.sergey.pisarev.model

class Rect : Point {
    var height = 0.0
    var width = 0.0

    constructor() {}
    constructor(point: Point, height: Double, width: Double) {
        x = point.x
        z = point.z
        this.height = height
        this.width = width
    }

    fun setRect(point: Point, height: Double, width: Double) {
        x = point.x
        z = point.z
        this.height = height
        this.width = width
    }

    fun setRect(x: Double, z: Double, height: Double, width: Double) {
        this.x = x
        this.z = z
        this.height = height
        this.width = width
    }

    fun isInsideRect(x: Double, z: Double): Boolean {
        return x >= this.x && x <= this.x + width && z >= this.z && z <= this.z + height
    }
}